openthread-tock/platform/platform-tock/flash.c

```diff
@@ -0,0 +1,73 @@
+// TODO COPYRIGHT
+
+#include <openthread/platform/flash.h>
+#include <string.h>
+
+const uint32_t SWAP_SIZE = 512;
+uint8_t swapPart0[512];
+uint8_t swapPart1[512];
+
+// NOTE: This is just a temporary version of this file. OpenThread will
+// not operate properly if it is unable to "save" the state. This provides
+// an in memory representation of "flash" that will be lost across power
+// cycles.
```

Comment:
I think we should do this.

Reply:
We have an implementation of this that uses non-volatile flash (which is accessible by all apps). @Samir-Rashid and @atar13 are working on implementing a non-volatile flash capsule with isolation.

For this PR, do we want to just use the memory representation? It seems placing a networkkey in flash accessible by all apps is antithetical to Tock's separation.

Reply:
I'm wary of open todos and a not quite complete port of openthread. This is by no means directed at you, or anyone specifically, but urgency/code updates have a way of, shall we say, diminishing after a tutorial finishes.